### PR TITLE
checks for username and username in card overflow fix

### LINKED ISSUE #7307: Issue in adding new user in users section
> [!NOTE]
> Maintainer Note: The issue was filed a considerable amount of time ago. Ensure that the issue is still relevant before working on solving it. 

Before we enter anything in input section of user it is showing user is available in empty string.

Screenshot 2024-02-28 at 7 46 39 PM

There is also some problem with validation of user.

### LINKED ISSUE #8883: Content overflows from the card in /users
When I was working on issue #7307 , I found this bug
When I added an extra user with my login, and visited /users with my login , I saw this small css bug where the name is overflowing

**To Reproduce**

1. Go to '/users'
2. Click on 2nd pagination of dev_districtadmin login (you can find it in any other login too)
3. See error (username overflows)

**Expected behavior**
Not to overflow outside the defined card

**Screenshots**
Image
Image

**Desktop (please complete the following information):**

- OS: windows
- Browser : chrome
- Version - latest

**Additional context**
I would like to take up this if approved

### PR DESCRIPTION
**Pull request by SwanandBhuskute**
## Proposed Changes

- Fixes #7307 
- Fixes #8883 
- now, username and password check will display like this before entering text in the field and then normal green-red behaviour will be done (7307)
image
image
image

- Username was overflowing from the card, so fixed it with simple css (8883)
image

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Release Notes

- **New Features**
	- Enhanced validation for ABHA address input, ensuring proper user input before submission.
	- Real-time password strength feedback in the Reset Password component.
	- Improved username and password validation in the User Add component.
	- Added password validation rules in the User Profile component.

- **Bug Fixes**
	- Refined error handling for user management functions, improving user feedback during actions like updating working hours and deleting users.
	- Adjusted layout handling for long usernames in the Manage Users component.

These updates enhance user experience by providing clearer validation messages and improving overall functionality.

### CODE REVIEW

File: src/components/Users/UserProfile.tsx [hunk 1]
```diff
@@ -494,8 +494,6 @@ export default function UserProfile() {
           refetchUser();
           Notification.Success({ msg: "Profile picture updated." });
           setEditAvatar(false);
-        } else {
-          onError();
```
Comment (Jacobjeevan) [inline src/components/Users/UserProfile.tsx]: Why is this removed? 🤔

Comment (SwanandBhuskute) [reply src/components/Users/UserProfile.tsx]: yes sir, actually this else was just added 20hrs ago, it shows in my git history
image

so I was getting errors while generating PR during git commit, so thats why the lines got removed unknowingly as it seemed up to date

but i will pull from develop and update the PR

Comment (SwanandBhuskute) [reply src/components/Users/UserProfile.tsx]: my branch is 3 commits behind the history says, it was not in the morning ig! I will update

Comment (SwanandBhuskute) [reply src/components/Users/UserProfile.tsx]: issue solved sir, merging and pulling was not working, so manually added else block, sry this time

Comment (Jacobjeevan) [reply src/components/Users/UserProfile.tsx]: Hmm, ok.

Note: As Bodhish mentioned before, you can just refer to us by our names 👍

**Review by Jacobjeevan — APPROVED** [review]
LGTM

**Comment by nihal467** [comment]
LGTM

**Comment by bodhish** [comment]
@coderabbitai full review

File: src/components/Users/UserProfile.tsx [hunk 2]
```diff
@@ -907,20 +907,24 @@ export default function UserProfile() {
                           {validateRule(
                             changePasswordForm.new_password_1?.length >= 8,
                             "Password should be atleast 8 characters long",
+                            !changePasswordForm.new_password_1,
```
[the remark this answers is not in the document]
Comment (SwanandBhuskute) [reply src/components/Users/UserProfile.tsx]: will improve and update

[the remark this answers is not in the document]
Comment (SwanandBhuskute) [reply src/components/Users/UserProfile.tsx]: I did the changes and then it is working as expected like earlier
so I applied similar logic for validateRule for checking username
so should I do it or keep as it is? @coderabbitai

File: src/components/Users/UserProfile.tsx [hunk 3]
```diff
@@ -193,6 +193,30 @@ export default function UserProfile() {
...
+      },
+      {
+        test: (p: string) => /\d/.test(p),
+        message: "Password should contain at least 1 number",
+      },
+    ];
+    return rules.map((rule) =>
+      validateRule(rule.test(password), rule.message, !password),
+    );
+  };
```
[the remark this answers is not in the document]
Comment (SwanandBhuskute) [reply src/components/Users/UserProfile.tsx:218]: no it was not there earlier, so not going to implement it @coderabbitai

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**